### PR TITLE
Launchpad: Clean up task-helper

### PR DESCRIPTION
**Pull request by gabrielcaires**
Related to #86647

## Proposed Changes
* Remove the ~700 lines from the task-helper `getEnhancedTasks` 
* Move the getEnhancedTasks to a separate file
* Migrate the missing dependencies to the context they are used.
* Transform siteInfoQueryArgs to a function

### Review instructions
I recommend using the commits to follow the changes. 

* I started moving the big switch from the task helper b2093cc0f6bb5b6d57cecf1cebb9d216c59a04de
* I moved to the correct context  https://github.com/Automattic/wp-calypso/pull/86960/commits/aa0498e61d490c2ebc9612a6f33630237427363e#diff-1af91985184f6d7b895325749a29ebd09bd8f1e4a570a3fa86cc9fb892416c3fL90-L92 
* I moved the siteInfoQueryArgs to a function so it can be shareable between tasks df60d3cc144d1bc99c950594bd20c3e8c848c9f2
* So I extract the getEnhancedTasks from task helper to a separate file. df60d3cc144d1bc99c950594bd20c3e8c848c9f2

### Testing instructions
Since all flows are already reviewed with the new code we can just run a smoke test in 1-2 of them to check if the new code is working, completing the flow.

* You can use the `setup/free` . More instructions are available https://github.com/Automattic/wp-calypso/pull/86432

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-definitions/index.ts [hunk 1]
```diff
@@ -28,24 +33,83 @@ const DEFINITIONS: TaskActionTable = {
...
+/**
+ * Some attributes of these enhanced tasks will soon be fetched through a WordPress REST
+ * API, making said enhancements here unnecessary ( Ex. title, subtitle, completed,
+ * subtitle, badge text, etc. ). This will allow us to access checklist and task information
+ * outside of the Calypso client.
+ *
+ * Please ensure that the enhancements you are adding here are attributes that couldn't be
+ * generated in the REST API
+ */
+export function getEnhancedTasks( {
```
Comment (gabrielcaires) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-definitions/index.ts:64]: Extracted from task-helper

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-definitions/plan/index.tsx [hunk 2]
```diff
@@ -52,13 +55,15 @@ const getPlanTaskSubtitle = (
...
-	} = context;
+	const { siteSlug, displayGlobalStylesWarning, globalStylesMinimumPlan, planCartItem, site } =
+		context;
+
+	const productSlug = planCartItem?.product_slug ?? site?.plan?.product_slug;
+
+	const isVideoPressFlowWithUnsupportedPlan =
+		isVideoPressFlow( flow ) && ! planHasFeature( productSlug as string, FEATURE_VIDEO_UPLOADS );
+
+	const shouldDisplayWarning = displayGlobalStylesWarning || isVideoPressFlowWithUnsupportedPlan;
```
Comment (gabrielcaires) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-definitions/plan/index.tsx:66]: These variables are only used inside this task, so I moved from task-helper to here.

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-definitions/post/index.tsx [hunk 3]
```diff
@@ -15,8 +15,8 @@ const getFirstPostPublished: TaskAction = ( task, flow, context ): Task => {
 			( task.completed && isBlogOnboardingFlow( flow || null ) ) ||
 			false,
 		calypso_path: ! isBlogOnboardingFlow( flow || null )
-			? `/post/${ siteInfoQueryArgs?.siteSlug }`
-			: addQueryArgs( `https://${ siteInfoQueryArgs?.siteSlug }/wp-admin/post-new.php`, {
+			? `/post/${ siteSlug }`
```
Comment (gabrielcaires) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-definitions/post/index.tsx:17]: It was using siteSlug from the siteInfoQueryArgs when the siteSlug was already available, so I used it.

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-definitions/test/index.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import { buildTask } from '../../test/lib/fixtures';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {};
+
+const queryClient = new QueryClient( {
+	defaultOptions: {},
+} );
+
+describe( 'getEnhancedTasks', () => {
```
Comment (gabrielcaires) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-definitions/test/index.ts:15]: Tests extracted from task-helper file

**Comment by daledupreez** [comment]
@gabrielcaires, I see this is both a Draft and in our Needs Review column on our board. Can you confirm whether you want an early review, or if the automation just pulled the issue into the Needs Review column? (It does that!)

**Comment by gabrielcaires** [comment]
> @gabrielcaires, I see this is both a Draft and in our Needs Review column on our board. Can you confirm whether you want an early review, or if the automation just pulled the issue into the Needs Review column? (It does that!)

@daledupreez 
I was unsure if I should move it to ready for review yet since we can't merge it, but early feedback is always good :D , so I updated the pr status.

**Comment by gabrielcaires** [comment]
Fixing the tests

**Review by valterlorran — APPROVED** [review]
LGTM!

**Pull request merged**